cmd/frontend/graphqlbackend/external_services_test.go

```diff
@@ -427,6 +427,24 @@ func TestExcludeRepoFromExternalService_ExcludedRepoExists_SameRepoIsNotExcluded
 	assert.Equal(t, expectedConfig, *cachedUpdate.Config)
 }
 
+func TestTrimHostFromRepoName(t *testing.T) {
+	testCases := map[string]struct {
+		inputRepoName       string
+		expectedTrimmedName string
+	}{
+		"repo has a hostname":               {inputRepoName: "github.com/sourcegraph/sourcegraph", expectedTrimmedName: "sourcegraph/sourcegraph"},
+		"repo doesn't have a hostname":      {inputRepoName: "sourcegraph/horsegraph", expectedTrimmedName: "sourcegraph/horsegraph"},
+		"non-hostname prefix isn't trimmed": {inputRepoName: "source/graph/horsegraph", expectedTrimmedName: "source/graph/horsegraph"},
```

Comment:
```suggestion
		"non-hostname prefix isn't trimmed": {inputRepoName: "source/graph/horsegraph", expectedTrimmedName: "source/graph/horsegraph"},
		
		"GitLab nested project is trimmed": {inputRepoName: "gitlab.com/source/graph/horsegraph", expectedTrimmedName: "source/graph/horsegraph"},
```

Something like this would be good to test. GitLab often has multiple levels of nesting.

Reply:
added!